tools/scalajsenv.js

```diff
@@ -761,21 +777,24 @@ ScalaJS.TypeData.prototype.initClass = function(
 initClass(
 //!endif
     internalNameObj, isInterface, fullName,
-    ancestors, isRawJSType, parentData, isInstance, isArrayOf) {
+    ancestors, typeTag, isRawJSType, parentData, isInstance, isArrayOf) {
   const internalName = ScalaJS.propertyName(internalNameObj);
 
   isInstance = isInstance || function(obj) {
-    return !!(obj && obj.$classData && obj.$classData.ancestors[internalName]);
+    let data = obj && ScalaJS.ClassData[obj.$typeTag];
+    return !!(data && (data.ancestors["indexOf"](typeTag) >= 0))
   };
 
   isArrayOf = isArrayOf || function(obj, depth) {
-    return !!(obj && obj.$classData && (obj.$classData.arrayDepth === depth)
-      && obj.$classData.arrayBase.ancestors[internalName])
+    let data = obj && ScalaJS.ClassData[obj.$typeTag];
```

Comment:
Same remark as above on how to get `data` (although here you can't rewrite as a "one-liner", I think)